src/bytes.rs

```diff
@@ -0,0 +1,71 @@
+//! 
+//! TODO: Use this temporarily until web3::types implemented no-std
+//!
+
+use rustc_hex::{FromHex, ToHex};
+use serde::de::{Error, Unexpected, Visitor};
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+#[cfg(any(feature = "default", feature = "std", test))]
+use std::fmt;
+#[cfg(all(feature = "no-std", not(test)))]
+use core::fmt;
+
+/// Raw bytes wrapper
+#[derive(Clone, Debug, Default, PartialEq, Eq, Hash)]
+pub struct Bytes(pub Vec<u8>);
+
+impl<T: Into<Vec<u8>>> From<T> for Bytes {
+    fn from(data: T) -> Self {
+        Bytes(data.into())
+    }
+}
+
+impl Serialize for Bytes {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let mut serialized = "0x".to_owned();
+        serialized.push_str(self.0.to_hex::<String>().as_ref());
+        serializer.serialize_str(serialized.as_ref())
+    }
+}
+
+impl<'a> Deserialize<'a> for Bytes {
+    fn deserialize<D>(deserializer: D) -> Result<Bytes, D::Error>
+    where
+        D: Deserializer<'a>,
+    {
+        deserializer.deserialize_identifier(BytesVisitor)
+    }
+}
+
+struct BytesVisitor;
+
+impl<'a> Visitor<'a> for BytesVisitor {
+    type Value = Bytes;
+
+    fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        write!(formatter, "a 0x-prefixed hex-encoded vector of bytes")
+    }
+
+    fn visit_str<E>(self, value: &str) -> Result<Self::Value, E>
+    where
+        E: Error,
+    {
+        if value.len() >= 2 && &value[0..2] == "0x" {
+            let bytes = FromHex::from_hex(&value[2..]).map_err(|e| Error::custom(format!("Invalid hex: {}", e)))?;
+            Ok(Bytes(bytes))
+        } else {
+            Err(Error::invalid_value(Unexpected::Str(value), &"0x prefix"))
```

Comment:
Is it worth adding support for non `0x` headed hex strings? The `FromHex` functions should be able to catch incorrectly formatted strings otherwise.

Reply:
I just copied `web3::types::bytes::Bytes`, so that's what was in there. Thankfully, [the spec](https://github.com/ethpm/ethpm-spec/blob/9440027b57f640e44922b6bfbdadac7c3a1b5616/spec/package.spec.json#L198) requires it in this format, so it's not a big deal.